### PR TITLE
Feature/search projects

### PR DESCRIPTION
**Pull request by rubenb994**
## Description
- Implemented search functionality for project overview

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Navigate to /projects/overview
2. Type something into the search project input
3. System filters the list based on the input

## Link to issue

Closes: #72

### CODE REVIEW

**Comment by Brend-Smits** [comment]
Can you also implement the search functionality on the home page with this pull request or do you think it should be a different branch? 
Also, please assign this PR to the Project Board on the top right side!

File: src/app/services/internal-search.service.ts [hunk 1]
```diff
@@ -1,25 +1,26 @@
-/*
```
Comment (Brend-Smits) [inline src/app/services/internal-search.service.ts:1]: License file was removed here

File: src/app/models/resources/internal-search-query.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export interface InternalSearchQuery {
```
Comment (Brend-Smits) [inline src/app/models/resources/internal-search-query.ts:1]: License file, please add it. Same goes for the other new files. There is no automated process for this yet unfortunately.

Comment (MaxKorlaar) [reply src/app/models/resources/internal-search-query.ts:1]: IntelliJ can do this for you 😏

Comment (Brend-Smits) [reply src/app/models/resources/internal-search-query.ts:1]: Are you getting paid by @Jetbrains ? ;)

**Comment by rubenb994** [comment]
> Can you also implement the search functionality on the home page with this pull request or do you think it should be a different branch?
I think it should be on a different branch and a different issue (this one was worded quite vaguely)

> Also, please assign this PR to the Project Board on the top right side!
Done 👍

**Comment by Brend-Smits** [comment]
@StijnGroenen I forgot to mention this but including the whole User object in the search results, is a data leak correct? 
If so, it should be removed from the frontend en from the backend models. 
I don't think we need to include the user/user-id at all(convince me otherwise).

**Comment by StijnGroenen** [comment]
> @StijnGroenen I forgot to mention this but including the whole User object in the search results, is a data leak correct?
> If so, it should be removed from the frontend en from the backend models.
> I don't think we need to include the user/user-id at all(convince me otherwise).

The back end does not return the full User model. It returns the UserResourseResult. Though this resource result does have a lot of information in it that should not be sent to the front end. This needs to be changed.
User is only necessary if the front end wants to display the user that added the project. In that case, a name and probably the profile URL will be enough. Otherwise, it can be removed completely.

**Comment by Brend-Smits** [comment]
> > @StijnGroenen I forgot to mention this but including the whole User object in the search results, is a data leak correct?
> > If so, it should be removed from the frontend en from the backend models.
> > I don't think we need to include the user/user-id at all(convince me otherwise).
> 
> The back end does not return the full User model. It returns the UserResourseResult. Though this resource result does have a lot of information in it that should not be sent to the front end. This needs to be changed.
> User is only necessary if the front end wants to display the user that added the project. In that case, a name and probably the profile URL will be enough. Otherwise, it can be removed completely.

Thank you for the clarification. For now the Frontend has no ambition to show user information in the search results. We will only show some user information on the project-detail page I reckon. 
Opened a new issue: https://github.com/DigitalExcellence/dex-backend/issues/129 to fix this in the backend. 

This pull request should be changed accordingly as well.

File: src/app/models/domain/search-result.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *   See the GNU Lesser General Public License for more details.
+ *   
+ *   You can find a copy of the GNU Lesser General Public License 
+ *   along with this program, in the LICENSE.md file in the root project directory.
+ *   If not, see https://www.gnu.org/licenses/lgpl-3.0.txt 
+ *  
+ */
+import { User } from "./user";
+
+export interface SearchResult {
```
Comment (jcoenen96) [inline src/app/models/domain/search-result.ts:21]: Please update the wiki, in the wiki user does not exist

Comment (rubenb994) [reply src/app/models/domain/search-result.ts:21]: What should be updated on the wiki?

**Comment by rubenb994** [comment]
> > > @StijnGroenen I forgot to mention this but including the whole User object in the search results, is a data leak correct?
> > > If so, it should be removed from the frontend en from the backend models.
> > > I don't think we need to include the user/user-id at all(convince me otherwise).
> > 
> > 
> > The back end does not return the full User model. It returns the UserResourseResult. Though this resource result does have a lot of information in it that should not be sent to the front end. This needs to be changed.
> > User is only necessary if the front end wants to display the user that added the project. In that case, a name and probably the profile URL will be enough. Otherwise, it can be removed completely.
> 
> Thank you for the clarification. For now the Frontend has no ambition to show user information in the search results. We will only show some user information on the project-detail page I reckon.
> Opened a new issue: [DigitalExcellence/dex-backend#129](https://github.com/DigitalExcellence/dex-backend/issues/129) to fix this in the backend.
> 
> This pull request should be changed accordingly as well.

So the user property should be dropped from the SearchResult interface?

**Comment by Brend-Smits** [comment]
> Are we dropping the user property?

Yes, for now we are. In the future we can look into adding it.

**Comment by rubenb994** [comment]
Issues should be resolved now. I've also improved my code by adding a debouceTime on the input in order to avoid a lot of API calls.

**Pull request merged**